src/dashboard/src/components/backlog/views.py

```diff
@@ -0,0 +1,216 @@
+                                           doc_type='transferfile', filters={'term': {'status': 'backlog'}})
+    try:
+        doc_type = 'transferfile' if file_mode else 'transfer'
+
+        hit_count = conn.search(index='transfers', doc_type=doc_type, body=query, search_type='count')['hits']['total']
+        hits = conn.search(
+            index='transfers',
+            doc_type=doc_type,
+            body=query,
+            from_=start,
+            size=page_size,
+            sort=order_by + ':' + sort_direction if order_by else ''
+        )
+
+    except Exception as e:
+        logger.exception('Error accessing index: {}'.format(e))
+        return HttpResponse('Error accessing index.')
+
+    return helpers.json_response({
+        'iTotalRecords': hit_count,
+        'iTotalDisplayRecords': hit_count,
+        'sEcho': int(request.GET.get('sEcho', 0)),  # It was recommended we convert sEcho to int to prevent XSS
+        'aaData': elasticSearchFunctions.augment_raw_search_results(hits)
+    })
+
+
+def delete_context(request, uuid):
+    """
+    Provide contextual information to the deletion request page.
+
+    :param request: The Django request object
+    :param uuid: The UUID of the package requested for deletion.
+    :return: The request context
+    """
+    prompt = 'Delete package?'
+    cancel_url = reverse('components.backlog.views.execute')
+    return RequestContext(request, {'action': 'Delete', 'prompt': prompt, 'cancel_url': cancel_url})
+
+
+@decorators.confirm_required('backlog/delete_request.html', delete_context)
+def delete(request, uuid):
+    """
+    Request deletion of a package from a backlog transfer
+
+    :param request: The Django request object
+    :param uuid: The UUID of the package requested for deletion.
+    :return: Redirects the user back to the backlog page
+    """
+    try:
+        reason_for_deletion = request.POST.get('reason_for_deletion', '')
+        response = storage_service.request_file_deletion(
+            uuid,
+            request.user.id,
+            request.user.email,
+            reason_for_deletion
+        )
+
+        messages.info(request, response['message'])
+        elasticSearchFunctions.connect_and_mark_backlog_deletion_requested(uuid)
+
+    except requests.exceptions.ConnectionError:
+        error_message = 'Unable to connect to storage server. Please contact your administrator.'
+        messages.warning(request, error_message)
+    except slumber.exceptions.HttpClientError:
+        raise Http404
+
+    return redirect('backlog_index')
+
+
+def download(request, uuid):
+    """
+    Download a package from a requested backlog transfer.
+
+    :param request: The Django request object
+    :param uuid: UUID for the transfer we're downloading the package from
+    :return: Respond with a TAR'd version of the requested package
+    """
+    return HttpResponseRedirect(storage_service.download_file_url(uuid))
```

Comment:
Point this at `filesystem_ajax.views.download_ss` instead since that one does streaming, and keeps all the SS-downloading in one place. Perhaps remove this and just use the filesystem_ajax one?